src/app/utils/query-url-sanitization.ts

```diff
@@ -1,67 +1,97 @@
-const keyIdRegex = /\(\'\<?\{?[ ?0-9a-zA-Z-]*\}?\>?\'\)/g;
-
-// Matches patterns like "(query=<key>)" or "(itemat=<key>,mode=<mode>)"
-const functionParamInitialRegex = /(?<=\=).*?(?=,)/g;
-const functionParamFinalRegex = /(?<=\=).[^,]*(?=\))/g;
-
 // Matches pattterns within quotes e.g "displayName: Gupta"
 const quotedTextRegex = /"([^"]*)"/g;
 
-// Matches PII patterns
-const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
-const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
-const guidRegex =
- /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
-
-
 export function sanitizeQueryUrl(url: string): string {
   url = decodeURIComponent(url);
 
   // Extract query string
-  const { search } = parseSampleUrl(url);
+  const { search, queryVersion, requestUrl } = parseSampleUrl(url);
   const queryString: string = sanitizeQueryParameters(search);
 
-  // Drop query string
-  let sanitizedUrl = url.split('?')[0];
+  let resourceUrl = requestUrl;
+  const sections = resourceUrl.split('/');
+  sections.forEach(sec => {
+    if (containsIdentifier(sec)) {
+      const index = sections.indexOf(sec);
+      const replacementItemWithPrefix = `{${sections[index - 1]}-id}`;
+      resourceUrl = resourceUrl.replace(sec, replacementItemWithPrefix);
+    }
+  });
+
+  return `${GRAPH_URL}/${queryVersion}/${resourceUrl}${queryString}`;
+}
+
+function sanitizeQueryParameters(queryString: string): string {
+  const params = queryString.split('&');
+  let result: string = '';
+  if (params.length) {
+    params.forEach(param => {
+      result += sanitizeQueryParameterValue(param) + '&';
+    });
+    result = result.slice(0, -1);
+  }
+  return result;
+}
 
-  // Normalize parameters in param=<arbitraryKey> format with param={value} format
-  sanitizedUrl = sanitizedUrl.replace(functionParamInitialRegex, '<value>');
-  sanitizedUrl = sanitizedUrl.replace(functionParamFinalRegex, '<value>');
+/*
+* A string contains an identifier if either:
+*   1. it is an alphanumeric string
+*   2. special characters are present
+*/
+export function containsIdentifier(segment: string): boolean {
+  return isAlphaNumericString(segment)
+  || isNumericString(segment)
+  || hasSpecialCharacters(segment);
+}
 
-  // Replace IDs and ID placeholders with generic {id}
-  sanitizedUrl = sanitizedUrl.replace(keyIdRegex, '/<id>');
-  sanitizedUrl = sanitizedUrl.replace(guidRegex, '<guid>');
-  sanitizedUrl = sanitizedUrl.replace(numberRegex, '<number>');
+export function hasSpecialCharacters(segment: string): boolean {
+  const specialCharacters = ['.', '=', '@', '-'];
+  return specialCharacters.some((character) => segment.includes(character));
+}
 
-  // Redact PII
-  sanitizedUrl = sanitizedUrl.replace(emailRegex, '<email>');
-  return `${sanitizedUrl}${queryString}` ;
+export function isNumericString(segment: string): boolean {
+  const numberRegex = /^[0-9]*$/;
```

Comment:
`^` is negation for selecting the character set. see regex101.com for testing the regular expressions.